demo/js/demo-animated-dialog.js

```diff
@@ -0,0 +1,84 @@
+const { useState } = require('react');
+const React = require('react');
+const ReactDOM = require('react-dom');
+const FocusTrap = require('../../dist/focus-trap-react');
+
+const container = document.getElementById('demo-animated-dialog');
+
+const focusTrapOptions = {
+  checkCanFocusTrap: (trapContainers) => {
+    const results = trapContainers.map((trapContainer) => {
+      return new Promise((resolve) => {
+        const interval = setInterval(() => {
+          if (getComputedStyle(trapContainer).visibility !== 'hidden') {
+            resolve();
+            clearInterval(interval);
+          }
+        }, 5);
+      });
+    });
+    // Return a promise that resolves when all the trap containers are able to receive focus
+    return Promise.all(results);
+  },
+  // Called after focus is sent to the focus trap
+  onPostActivate: () => {
+    // eslint-disable-next-line no-console
+    console.log('Focus has been sent to the animated focus trap');
+  },
+};
+
+const DemoAnimatedDialog = () => {
+  const [isTrapActive, setIsTrapActive] = useState(false);
+
+  return (
+    <>
+      <style>{`
```

Comment:
I know that you are going to ask me to move this to the CSS style sheet. I want to keep this here though.

Unlike with the core `focus-trap` demos, the react demos are able to show everything relevant to recreating the demo in the one file nice the HTML is also stored in this JS file.

The styles for the animated dialogs are an important aspect of the demo so they should be displayed as part of the demo and not hidden in the main css file.